editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/util/maven/MavenUtils.java

```diff
@@ -157,6 +158,7 @@ public static void ensureRepositoryExists(List<Repository> repositories, String
 			repo.setUrl(repoURI);
 			repositories.add(repo);
 		}
+		return !exists;
```

Comment:
Why?
if the repository has not been found in the first loop, 'exists' value is false. Then the return value is true.